R/utils-get_code_dependency.R

```diff
@@ -0,0 +1,498 @@
+
+  lapply(
+    lapply(graph, paste, collapse = "***"),
+    function(x) {
+      split <- strsplit(x, split = ":", fixed = TRUE)[[1]]
+      deps <- strsplit(split[1], split = "***", fixed = TRUE)[[1]]
+
+      if (length(split) == 1) {
+        unique(deps)
+      } else if (length(split) == 2) {
+        infl <- strsplit(split[2], split = "***", fixed = TRUE)[[1]][-1]
+        c(unique(deps), ":", unique(infl))
+      }
+    }
+  )
+
+}
+
+# graph_parser ----
+
+#' Return the indexes of calls of code needed to reproduce the object
+#'
+#' @param x The name of the object to return code for.
+#' @param graph A result of `code_graph()`.
+#' @param skip `NULL` or `character` vector. In a recursive call, it is needed to drop parent object to omit dependency
+#' cycles.
+#'
+#' @return `numeric` vector indicating which calls of `graph` are required to build the object passed by name in `x`.
+#'
+#' @keywords internal
+#' @noRd
+graph_parser <- function(x, graph, skip = NULL) {
+
+  skip <- c(x, skip)
+
+  occurence <-
+    vapply(graph, function(call) {
+      if (":" %in% call) {
+        call <- call[1:(which(":" == call)-1)]
+      }
+      x %in% call
+    },
+    logical(1)
+  )
+
+  influencers <-
+    unlist(
+      lapply(graph[occurence], function(call) {
+        if (":" %in% call) {
+          call[(which(":" == call)+1):length(call)]
+        }
+      })
+    )
+  influencers <- setdiff(influencers, skip)
+
+  if (length(influencers)) {
+    influencers_ids <-
+      lapply(influencers, function(influencer){
+        graph_parser(influencer, graph[1:max(which(occurence))], skip)
+      })
+
+    sort(unique(c(which(occurence), unlist(influencers_ids))))
+  } else {
+    which(occurence)
+  }
+
+}
+
+# utils -----------------------------------------------------------------------------------------------------------
+
+#' @keywords internal
+#' @noMd
+assert_classes <- function(code, names) {
+  checkmate::assert_multi_class(code, classes = c("character", "expression"))
+  checkmate::assert_character(names)
+}
+
+#' @keywords internal
+#' @noMd
+assert_code <- function(code) {
+  if (is.expression(code)) {
+    if (!is.null(attr(code, "srcref"))) {
+      parsed_code <- code
+    } else {
+      stop("The 'expression' code input does not contain 'srcref' attribute.")
+    }
+  }
+
+  if (is.character(code)) {
+    parsed_code <- parse(text = code, keep.source = TRUE)
+  }
+  parsed_code
+}
+
+#' @keywords internal
+#' @noMd
+assert_names <- function(names, pd) {
+  symbols <- unique(pd[pd$token == "SYMBOL", "text"])
+  if (!all(names %in% symbols)) {
+    warning(
+      "Object(s) not found in code: ",
+      toString(setdiff(names, symbols))
+    )
+  }
+}
+
+#' @keywords internal
+#' @noMd
+is_empty <- function(code){
+  identical(code, character(0)) || identical(trimws(code), "")
+}
+
+# # # examples and test -----------------------------------------------------------------------------------------------
```

Comment:
please remove commented text

Reply:
Yes, this will be removed before the merge : ) It's mainly here for easier review